jac/jaclang/compiler/parser.py

```diff
@@ -760,51 +756,29 @@ def enum_stmt(self, kid: list[ast.AstNode]) -> ast.EnumBlockStmt:
                     | abstract_ability
                     | ability
             """
-            if isinstance(kid[0], (ast.PyInlineCode, ast.Ability)):
-                return kid[0]
-            elif isinstance(kid[0], (ast.Name)):
-                if (
-                    len(kid) >= 3
-                    and isinstance(kid[-1], ast.Expr)
-                    and not isinstance(kid[-1], ast.String)
-                ):
-                    semstr = (
-                        kid[2]
-                        if len(kid) > 3 and isinstance(kid[2], ast.String)
-                        else None
-                    )
-                    targ = ast.SubNodeList[ast.Expr](
-                        items=[kid[0]], delim=Tok.COMMA, kid=[kid[0]]
-                    )
-                    kid[0] = targ
-                    return ast.Assignment(
-                        target=targ,
-                        value=kid[-1],
-                        type_tag=None,
-                        kid=kid,
-                        semstr=semstr,
-                        is_enum_stmt=True,
-                    )
-                else:
-                    semstr = (
-                        kid[2]
-                        if len(kid) == 3 and isinstance(kid[2], ast.String)
-                        else None
-                    )
-                    targ = ast.SubNodeList[ast.Expr](
-                        items=[kid[0]], delim=Tok.COMMA, kid=[kid[0]]
-                    )
-                    kid[0] = targ
-                    return ast.Assignment(
-                        target=targ,
-                        value=None,
-                        type_tag=None,
-                        kid=kid,
-                        semstr=semstr,
-                        is_enum_stmt=True,
-                    )
-            elif isinstance(kid[0], (ast.PyInlineCode, ast.ModuleCode)):
-                return kid[0]
+            if stmt := (
+                self.match(ast.PyInlineCode)
+                or self.match(ast.ModuleCode)
+                or self.match(ast.Ability)
+            ):
+                return stmt
+            elif name := self.consume(ast.Name):
```

Comment:
The above if always return so this `elif` is not necessary. remove the elif.

```py
if condition:
  return something()

elif name := consume(...):
  do_something()
```

can be written as:
```py
if condition:
  return something()

name = consume(...):
do_something()
```